exception handling for jp2 creation

p-galligan opened the pull request:
Adds a try and exception catch to running the subprocess. If there's a CalledProcessError, raise a generic exception pointing to the file.

Pretty simple solution that'll immediately stop processing on the first failure.

Fixes #82

Commit:
exception handling for jp2 creation

Adds a try and exception catch to running the subprocess. If there's a CalledProcessError, raise a generic exception pointing to the file.

p-galligan commented:
I wasn't completely sure how to test the memory usage of this, which is the only potential worry that I have.

helrond reviewed (changes requested):
I think I would suggest _not_ using try/except here, because you're just re-raising a more generic exception (I'm also not sure that we can assume the exception will be caused by a faulty TIFF so that message may be misleading).

p-galligan commented:
> I think I would suggest _not_ using try/except here, because you're just re-raising a more generic exception (I'm also not sure that we can assume the exception will be caused by a faulty TIFF so that message may be misleading).

So, two things: 

1. I can capture the exact error message from the subprocess failing, if that's more helpful.
2. Maybe I'm just being slow today, but I'm not sure if there's a faster way to get to the correct state without using try and except because a failure with `check=True` immediately ends the code block and anything after it will not run. IMO the cleanest would be to check for the CalledProcessError and then raise the error that comes with that.

I could do something like the below, but it seems a little repetitive:

```
exit_status = subprocess.run(cmd, check=True)
if exit_status.returncode != 0:
    raise Exception
else:
   subprocess.run(cmd)
```

p-galligan commented:
The above may also cause the same problems as `check_output` because it's basically doing the same thing.

helrond commented:
Catching and re-raising in that block is unnecessary because `create_jp2` is called in a try/except block [here](https://github.com/RockefellerArchiveCenter/iiif-pipeline/blob/base/iiif_pipeline/pipeline.py#L65). Any exceptions thrown by the `create_jp2` method will be caught there.

p-galligan commented:
> Catching and re-raising in that block is unnecessary because `create_jp2` is called in a try/except block [here](https://github.com/RockefellerArchiveCenter/iiif-pipeline/blob/base/iiif_pipeline/pipeline.py#L65). Any exceptions thrown by the `create_jp2` method will be caught there.

Ahhh, I see what you're saying. Yeah, so it totally catches the output from the failed subprocess check.

For example: 

```
Error processing identifier fFXsb43H6P28AnTp3Sp7sj with ref_id c096b4e939f5444c9ba791134cb0808d: Command '['/usr/local/bin/opj_compress', '-i', '/source_files/c096b4e939f5444c9ba791134cb0808d/master/rac_rfdiaries_12-2_miller_1932-1959_022_001.tif', '-o', '/tmp/images/fFXsb43H6P28AnTp3Sp7sj_001.jp2', '-n', '7', '-SOP', '-r', '1.5', '-c', '[256,256],[256,256],[128,128]', '-b', '64,64', '-p', 'RPCL']' returned non-zero exit status 1.
```

@bonniegee Is that enough information for you? If we wanted the specific error message like `tiftoimage: Bad color format 3. Only RGB(A) and GRAY(A) has been implemented. Aborting. Unable to load tiff file`, we'd have to basically replicate `check_output` functionality as far as I understand.

https://stackoverflow.com/a/4760517

bonniegee commented:
@p-galligan I think that error is enough!

Commit:
remove try/except

Removes try except process and lets the error handling in `pipeline.py` catch the exception.
